### PR TITLE
Metrics for merge time 

### PR DESCRIPTION
**Pull request by hvilloria**
## What does this PR do?
It uses MergeTime model as parameter to calculate metrics for every entity. This fixes the merge time average metrics bug. 

## Hosward notes
* this branch was initially created to resolve the issue for week metrics values but i could not finish it
* the main idea was to refactor the calculation because i thought the independency between de distribution and the metric calculation was the problem but it was not.
* after the refactor all the information i collected is in this [Jira](https://rootstrap.atlassian.net/browse/EM-155), i hope it be helpful

### CODE REVIEW

File: spec/services/metrics/merge_time/per_department_spec.rb [hunk 1]
```diff
@@ -2,44 +2,93 @@
...
       before do
-        create(:metric, value: 30.minutes, ownable: first_project, name: :merge_time)
-        create(:metric, value: 25.minutes, ownable: second_project, name: :merge_time)
+        first_project_pull_request.update!(merged_at: beginning_of_day + 1.hour)
+        second_project_pull_request.update!(merged_at: beginning_of_day + 2.hours)
       end
 
       it 'creates two metrics' do
-        expect { described_class.call }.to change { Metric.count }.from(2).to(4)
+        expect { described_class.call }.to change { Metric.count }.from(0).to(2)
```
Comment (santib) [inline spec/services/metrics/merge_time/per_department_spec.rb:34]:
```suggestion
        expect { described_class.call }.to change { Metric.count }.by(2)
```
?

**Review by santib — APPROVED** [review]
Great job 💯

File: app/services/metrics/merge_time/per_user_project.rb [hunk 2]
```diff
@@ -1,42 +1,38 @@
...
           end
-          calculate_metrics_avg(entities, :merge_time)
         end
       end
 
-      def pull_requests_in_batches
-        filtered_pull_requests.find_each(batch_size: BATCH_SIZE).lazy.each do |pull_request|
-          yield(pull_request)
+      def merge_times_in_batches
+        filtered_merge_times.each do |project_id, owner_id, merge_time_value|
```
Comment (hdamico) [inline app/services/metrics/merge_time/per_user_project.rb:17]: Why did you stop using BATCH_SIZE and `lazy`?

**Review by hdamico — APPROVED** [review]

File: app/services/metrics/merge_time/per_user_project.rb (hunk 2 above)
Comment (SandroDamilano) [reply app/services/metrics/merge_time/per_user_project.rb:17]: I don't have an answer for that haha
@hvilloria do you? Otherwise I'll just put them back

Comment (SandroDamilano) [reply app/services/metrics/merge_time/per_user_project.rb:17]: Sorry, I do have an answer.
`filtered_merge_times` no longer returns an `ActiveRecord_Relation`, it returns an `Array`. So we can't use `find_each` anymore.
Given this, I'll just delete this `merge_times_in_batches` method, as it's useless now

**Pull request merged**